src/tools/crawling_tools.py

```
@@ -516,132 +674,202 @@ async def crawl_dir(
-        
-        # Report initial progress
-        await ctx.report_progress(progress=0, total=100, message="Starting directory crawl...")
-        
-        await ctx.report_progress(progress=10, total=100, message="Scanning directory for files...")
-        
-        # Crawl directory to get file contents
+    # Initialize error handler and enhanced logging
+    error_handler = MCPErrorHandler("crawl_dir", operation_id)
+    mcp_logger = OperationMonitor().get_logger("crawl_dir")
+    
+    async with log_operation(ctx, "crawl_dir", operation_id, {"directory_path": directory_path, "max_files": max_files}) as op_logger:
```

Ruff (0.12.2)

Local variable `op_logger` is assigned to but never used

Remove assignment to unused variable `op_logger`

(F841)